Marketplace/Cardz/Card.js

```diff
@@ -0,0 +1,296 @@
+        var properties = JSON.parse(props.userData);
+        var held = properties.held;
+        if (held == true) {
+            //get hand
+            hand = args[0];
+            //delete an overlay if there is one
+            try {
+                Overlays.deleteOverlay(cardOverlay);
+                cardOverlay = undefined;
+            } catch (err) {
+                //e
+            }
+        } else if (held == false) {
+            hand = args[0];
+            //unparent to hand and change held state
+            var changeTexture = {
+                parentID: "",
+                parentJointIndex: "",
+                textures: '{ "HiddenCardFile": "https://hifi-content.s3.amazonaws.com/jedon/Game_Creater_Toolkit/DeckOfCards/DeckOfCardsTexture/CARD_X.jpg"}',
+            };
+            Entities.editEntity(_this.entityID, changeTexture);
+            var cardChannel = "card-channel-".concat(deckHandlerID);
+            var data = [false, _this.entityID, MyAvatar.sessionUUID, hand, _this.entityID];
+            Messages.sendMessage(cardChannel, JSON.stringify(data));
+        }
+    };
+
+    _this.startDistanceGrab = function(entityID, args) {
+        //if the glove is in your hand then check to see if its touching a card
+        var props = Entities.getEntityProperties(_this.entityID);
+        var properties = JSON.parse(props.userData);
+        var held = properties.held;
+        if (held == true) {
+            //get hand
+            hand = args[0];
+            //delete an overlay if there is one
+            try {
+                Overlays.deleteOverlay(cardOverlay);
+                cardOverlay = undefined;
+            } catch (err) {
+                //e
+            }
+        } else if (held == false) {
+            hand = args[0];
+            //unparent to hand and change held state
+            var changeTexture = {
+                parentID: "",
+                parentJointIndex: "",
+                textures: '{ "HiddenCardFile": "https://hifi-content.s3.amazonaws.com/jedon/Game_Creater_Toolkit/DeckOfCards/DeckOfCardsTexture/CARD_X.jpg"}',
+            };
+            Entities.editEntity(_this.entityID, changeTexture);
+            var cardChannel = "card-channel-".concat(deckHandlerID);
+            var data = [false, _this.entityID, MyAvatar.sessionUUID, hand, _this.entityID];
+            Messages.sendMessage(cardChannel, JSON.stringify(data));
+        }
+    };
+
+    _this.releaseGrab = function (entityID, args) {
+        var hand = args[0];
+        //check if the object you are holding is a card and if it isnt already in your hand
+        var props = Entities.getEntityProperties(_this.entityID);
+        var properties = JSON.parse(props.userData);
+        var checkIfCard = properties.card;
+        var held = properties.held;
+        var deckHandlerID = properties.deckHandlerID;
+        if ((checkIfCard == true) && (held == false) && (hand == "right")) {

```

Comment:
Please refactor this so there isn't as much duplicate code in this file. There is a lot of shared code between this function and the next.